plugins/moz_databricks.py

```diff
@@ -0,0 +1,171 @@
+            autoscaling in dev
+        :param enable_autoscale: Enable autoscaling for the job
+        :param disable_on_dev: Turn the job into a no-op if run in development
+        :param release_label: Databricks Runtime versions,
+            run `databricks clusters spark-versions` for possible values.
+        :param iam_role: An Amazon Resource Name (ARN) specifying an iam role
+        :param owner: The e-mail address of the user owning the job.
+        :param uri: argument from EMRSparkOperator for compatibility
+        :param output_visibility: argument from EMRSparkOperator for compatibility
+
+        :param kwargs: Keyword arguments to pass to DatabricksSubmitRunOperator
+        """
+
+        if enable_autoscale:
+            if not max_instance_count:
+                raise ValueError("`max_instance_count` should be set when "
+                                 "`enable_autoscale` is enabled.")
+            if (max_instance_count < instance_count or
+                    dev_max_instance_count < dev_instance_count):
+                raise ValueError("The max instance count should be greater "
+                                 "than the instance count.")
+
+        is_dev = self.deploy_environment == 'dev'
+        self.disable_on_dev = disable_on_dev
+        self.job_name = job_name
+
+        jar_task = None
+        python_task = None
+        libraries = []
+
+        # Create the cluster configuration
+        new_cluster = {
+            "spark_version": release_label,
+            "node_type_id": self.instance_type,
+            "aws_attributes": {
+                "availability": "ON_DEMAND",
+                "instance_profile_arn": iam_role or self.default_iam
+
+            },
+            "spark_env_vars": env,
+            "custom_tags": {
+                "Owner": owner,
+                "Application": "databricks",
+                "Source": "Airflow",
+                "Job": job_name,
+            }
+        }
+
+        min_workers = dev_instance_count if is_dev else instance_count
+        max_workers = dev_max_instance_count if is_dev else max_instance_count
+        if enable_autoscale:
+            new_cluster["autoscale"] = {
+                "min_workers": min_workers,
+                "max_workers": max_workers,
+            }
+        else:
+            new_cluster["num_workers"] = min_workers
+
+        # Parse the environment variables to bootstrap the tbv/mozetl workflow
+        if env.get("TBV_CLASS"):
+            jar_task = {
+                "main_class_name": env["TBV_CLASS"],
+                # Reconstruct TBV parameters from the environment, scallop does
+                # not support reading arguments in this form
+                "parameters": sum([
+                    ["--{}".format(key[4:].replace("_", "-")), value]
+                    for key, value in env.items()
+                    if key.startswith("TBV_") and key != "TBV_CLASS"
```

Comment:
I understand the impetus to try to make this class as backwards compatible as possible, but I think we should consider passing parameters through without tbv_envvar if we can avoid this re-munging things we just munged earlier. I'm not sure this is worth the interface compatibility

Reply:
I don't think this operator is worth the interface incompatibility because I think the environment is the right level to think about the job calling convention. For example, if one day we migrated our jobs to submit via a Docker master, the easiest way to pass arguments would be through the environment. The `tbv_envvar` and `mozetl_envvar` interface with the corresponding runners have been working well, so far. 

It's a bit convoluted to do this extra work when we could pass in the raw string, but that's the price for choosing environment variables for passing context between run-times. My preferred solution to this would be [environment variable support like click](http://click.pocoo.org/5/options/#values-from-environment-variables).

Also since the interface is compatible, I also think it would be cool to monkey-patch EMRSparkOperator with the MozDatabricksRunSubmitOperator later.